jsftemplating/src/main/java/com/sun/jsftemplating/el/PageSessionResolver.java

```diff
@@ -97,20 +136,7 @@ public Class<?> getType(ELContext elContext, Object base, Object property) {
 
     @Override
     public void setValue(ELContext elContext, Object base, Object property, Object value) {
-        if (base != null) {
```

Comment:
How does this work? How does this set a page context value? Before, this method expected `base == null` and `property != null` and set the property into the page context. This means I think that if a page set some variable (with no base), the variable was added to the page context.

The method after your change expects that `base != null`, which is different to the condition before. And it doesn't set anything in the page context. What happens if a page just defines a new variable (without base) ?

Reply:
Prior to Faces 4, `PageSessionResolver` was `VariableResolver`, not `ELResolver` and therefore did not setup any values.

All page session values are set with our custom handlers, e.g. `setPageSessionAttribute`.

Now we first try to read updated values, which come to us from standard scopes, and if its not found, read from page session map.

Reply:
OK, I understand, thanks :)

Reply:
@OndroMih, page session attributes are "read-only" from the point of view EL engine, imo. Here description from spec https://jakarta.ee/specifications/faces/3.0/jakarta-faces-3.0#a2773.

I tried to implement this part in the `getValue()` method

> Get the head of the VariableResolver chain and call resolveVariable(facesContext, property) and return the result

by checking standard scopes before our custom page session scope. This should prevent us from reading an obsoleted values.

Updates of the values from requests handled by Faces `ScopedAttributeELResolver`.